docs/staging-use-cases.md

```diff
@@ -4,22 +4,42 @@ creation-date: 2019-04-16
 last-updated: 2019-04-16
 ---
 
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
```

Comment:
should doctoc be added as update/verify scripts under:
https://github.com/kubernetes-sigs/cluster-api/tree/master/hack
and be part of CI?

for example,, if anyone makes changes to the document they would also need to run e.g. ./hack/update-generated-docs.sh, otherwise ./hack/verify-generated-docs.sh would not pass CI.

Reply:
Thanks @neolit123 , but not all of the document using `doctoc` go generate toc in this repo, how about open an issue to have some discussion for this? And for this PR, how about merge it first?

Reply:
the doctoc verify/update can be scoped to work only on selected files.
+1 opening a issue, so that others can reproduce the ToC.
thanks.

Reply:
Opened an issue at https://github.com/kubernetes-sigs/cluster-api/issues/934